### PR TITLE
Bugfix/better routing auth failure

### PR DESCRIPTION
**Pull request by KaleabTessera**
Implementation of https://github.com/deep-learning-indaba/Baobab/issues/180 .

### CODE REVIEW

File: api/app/users/tests.py [hunk 1]
```diff
@@ -274,7 +276,7 @@ def test_password_reset(self):
             'code': "bad code",
             'password': 'abc123'
         })
-        assert response.status_code == 401
+        assert response.status_code == 418
```
Comment (avishkar58) [inline api/app/users/tests.py]: LOL

File: webapp/src/services/base.service.js [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+  },
+  function(error) {
+    if (
+      error &&
+      error.response &&
+      error.response.status === 401 &&
+      error.response.data &&
+      error.response.data.type === "UNAUTHORIZED"
+    ) {
+      localStorage.removeItem("user");
```
Comment (avishkar58) [inline webapp/src/services/base.service.js]: We also need to signal (somehow?) to the UserDropdown component that the user has been logged out...

Comment (KaleabTessera) [reply webapp/src/services/base.service.js]: Currently working on this, it is a little more complicated than I expected.

Comment (KaleabTessera) [reply webapp/src/services/base.service.js]: So I sorted this issue. 

I tried adding a listener to localstorage changes and then this updates user in the User.js, but the listener wasn't been triggered from changes to localstorage from axios. 

I found a work around, but just always checking localstorage and comparing it to the props received.

Quite a trick problem, redux was made to solve these kinds of issues, with communicating state changes to a wide variety of components, but it works so it fine for now :)

**Pull request merged**